src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/F1R2CountsCollector.java

```diff
@@ -0,0 +1,216 @@
+package org.broadinstitute.hellbender.tools.walkers.readorientation;
+
+import com.google.common.primitives.Ints;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.metrics.StringHeader;
+import htsjdk.samtools.util.Histogram;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Nucleotide;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class F1R2CountsCollector {
+
+    public static final String ALT_TABLE_EXTENSION = ".alt_table";
+    public static final String ALT_HIST_EXTENSION = ".alt_histogram";
+    public static final String REF_HIST_EXTENSION = ".ref_histogram";
+
+    private static final Logger logger = LogManager.getLogger(F1R2CountsCollector.class);
+
+    private final Set<String> samples;
+
+    private final SAMFileHeader header;
+
+    private final CollectF1R2CountsArgumentCollection CF1R2Args;
+
+    // For each sample and for reference context, count ref sites in a histogram keyed by depth
+    private final Map<String, Map<String, Histogram<Integer>>> refSiteHistograms;
+
+    // For each sample store the total depths of alt sites with alt depth = 1 separately to save memory
+    private Map<String, DepthOneHistograms> depthOneAltHistograms;
+
+    // alt table writer for each sample
+    private Map<String, AltSiteRecord.AltSiteRecordTableWriter> altTableWriters;
+
+    private final File outputTarGzFile;
+
+    private final File tmpDir = IOUtils.createTempDir("untarred");
+
+    public F1R2CountsCollector(final CollectF1R2CountsArgumentCollection CF1R2Args, final SAMFileHeader header, final File outputTarGzFile) {
+        this.CF1R2Args = CF1R2Args;
+        samples = ReadUtils.getSamplesFromHeader(header);
```

Comment:
We should limit the samples to tumor samples here. The collector stores normal samples in `samples`, but the actual data collection is done only on the tumor samples, which down the line creates an empty prior table for the normal samples in `LearnReadOrientationModel.`

Reply:
done